execution/binary/scalar.go

```diff
@@ -120,8 +121,11 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 		}
 		out = append(out, step)
 		o.next.GetPool().PutStepVector(vector)
+		o.scalar.GetPool().PutStepVector(scalarIn[v])
```

Comment:
can this be out of bounds?

Reply:
You're right I missed that. It should be `len(scalarIn) > v`

Reply:
I think ideally we want to release these vectors independently. The scalar can also have more elements in the batch, and in this case that memory will never be released. So maybe we can add another loop for the scalar vectors?

Reply:
That makes sense! Done in https://github.com/thanos-community/promql-engine/pull/98/commits/14896cacd44341a0b473ac2767bccc2ef108e529